packages/api/src/controllers/room.test.ts

```diff
@@ -64,6 +67,60 @@ describe("controllers/room", () => {
       const getRoomRes = await res.json();
       expect(getRoomRes.id).toBe(roomRes.id);
       expect(getRoomRes.userId).toBeUndefined();
+
+      const mockDeleteRoom = jest.spyOn(roomServiceClient, "deleteRoom");
+      res = await client.delete(`/room/${roomRes.id}`);
+      expect(res.status).toBe(204);
+      expect(mockDeleteRoom).toHaveBeenCalledTimes(1);
+      expect(mockDeleteRoom.mock.calls[0][0]).toBe(roomRes.id);
+
+      // should no longer exist
+      res = await client.get(`/room/${roomRes.id}`);
+      expect(res.status).toBe(404);
+    });
+
+    it("should start and stop egress", async () => {
+      let res = await client.post(`/room`);
+      expect(res.status).toBe(201);
+      const roomRes = await res.json();
+      expect(roomRes.id).toBeDefined();
+
+      let mockStartEgress = jest.spyOn(
+        egressClient,
+        "startRoomCompositeEgress"
+      );
+      mockStartEgress.mockReturnValueOnce(
+        Promise.resolve({
+          egressId: "egress-id",
+        })
+      );
+      let mockStopEgress = jest.spyOn(egressClient, "stopEgress");
+      mockStopEgress.mockReturnValueOnce(Promise.resolve(undefined));
+
+      res = await client.delete(`/room/${roomRes.id}/egress`);
+      expect(res.status).toBe(400);
+
+      res = await client.post(`/room/${roomRes.id}/egress`, {
+        rtmpURL: "rtmp",
+      });
+      expect(res.status).toBe(204);
+
+      // already started so should 400
+      res = await client.post(`/room/${roomRes.id}/egress`, {
+        rtmpURL: "rtmp",
+      });
+      expect(res.status).toBe(400);
```

Comment:
Should it allow multiple (different) egresses to be added?

Reply:
We don't need that for the current design so for now at least I think we should keep it simple and just allow one.